code/game/objects/structures/headpike.dm

```diff
@@ -13,9 +16,15 @@
 /obj/structure/headpike/CheckParts(list/parts_list)
 	..()
 	var/obj/item/bodypart/head/H = locate() in contents
-	if(H)
-		update_icon()
-		name = "[H.real_name]'s head on a spear"
+	var/obj/item/twohanded/spear/S = locate() in contents
+	var/obj/item/twohanded/bonespear/BS = locate() in contents
```

Comment:
Can't you take these from the parts list?

Reply:
As far as I was aware of, that was what I was doing.

Reply:
You're currently checking in the item's contents for the parts, which does work but isn't very efficient. The proc takes in a list of parts used for the recipe (parts_list), in this case a spear and a head; you could sort them with an istype then assign them to their proper vars.

Reply:
Is there an example for this? I'm not exactly sure how to go about this method.